File: src/test/java/aoc/Day01Test.java
package aoc;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import testutils.Day;
import testutils.TestFilePathParameterResolver;

import static org.junit.jupiter.api.Assertions.*;

@Day("01")
@ExtendWith(TestFilePathParameterResolver.class)
class Day01Test {

    private final Day01 day01 = new Day01();

    @Test
    void part1(String sample, String test) {
        final var testResult = day01.part1(sample);
        System.out.println("testResult = " + testResult);
        assertEquals(24_000, testResult);

        final var result = day01.part1(test);
        System.out.println("result = " + result);
        assertEquals(71_124, result);
    }

    @Test
    void part2(String sample, String test) {
        final var testResult = day01.part2(sample);
        System.out.println("testResult = " + testResult);
        assertEquals(45_000, testResult);

        final var result = day01.part2(test);
        System.out.println("result = " + result);
        assertEquals(204_639, result);
    }
}

File: src/test/java-kotlin/aoc/Day01Test.kt
package aoc

import org.junit.jupiter.api.Test
import utils.assertEqualTo

internal class Day01Test {

    @Test
    fun testSumDepthIncreases() = Day01.part1().also(::println).assertEqualTo(1_722)

    @Test
    fun testSumDepthRangeIncreases() = Day01.part2().also(::println).assertEqualTo(1_748)

    @Test
    fun `part 2 v2 gives same answer`() {
        val filePath = "input/day1.txt"
        Day01.part2(filePath).assertEqualTo(Day01.part2V2(filePath))
    }
}
